test/unit/reductions/exponentiated_gradient/test_exponentiatedgradient_arguments.py

```diff
@@ -35,35 +36,83 @@ class TestExponentiatedGradientArguments:
     @pytest.mark.parametrize("transformX", candidate_X_transforms)
     @pytest.mark.parametrize("A_two_dim", [False, True])
     @pytest.mark.uncollect_if(func=is_invalid_transformation)
-    def test_argument_types(self, transformX, transformY, transformA, A_two_dim):
+    def test_argument_types_difference_bound(self, transformX, transformY, transformA, A_two_dim):
         # This is an expanded-out version of one of the smoke tests
         X, y, A = _get_data(A_two_dim)
         merged_A = _map_into_single_column(A)
 
         transformed_X = transformX(X)
         transformed_y = transformY(y)
         transformed_A = transformA(A)
+        eps = 0.1
 
         expgrad = ExponentiatedGradient(
             LeastSquaresBinaryClassifierLearner(),
             constraints=DemographicParity(),
-            eps=0.1)
+            eps=eps)
         expgrad.fit(transformed_X, transformed_y, sensitive_features=transformed_A)
 
         def Q(X): return expgrad._pmf_predict(X)[:, 1]
         n_predictors = len(expgrad.predictors_)
 
-        disparity_moment = DemographicParity()
+        disparity_moment = DemographicParity(difference_bound=eps)
         disparity_moment.load_data(X, y, sensitive_features=merged_A)
         error = ErrorRate()
         error.load_data(X, y, sensitive_features=merged_A)
         disparity = disparity_moment.gamma(Q).max()
+        disp = disparity_moment.gamma(Q)
+        disp_eps = disparity_moment.gamma(Q) - disparity_moment.bound()
         error = error.gamma(Q)[0]
 
         assert expgrad.best_gap_ == pytest.approx(0.0000, abs=_PRECISION)
         assert expgrad.last_iter_ == 5
         assert expgrad.best_iter_ == 5
         assert disparity == pytest.approx(0.1, abs=_PRECISION)
+        assert (np.all(np.isclose(disp - eps, disp_eps)))
+        assert error == pytest.approx(0.25, abs=_PRECISION)
+        assert expgrad._n_oracle_calls == 32
```

Comment:
this needs to be n_oracle_calls_ (trailing underscore) due to my change from yesterday (apologies, should have realized this would affect you and told you earlier today!)